main.go

```diff
@@ -73,23 +76,64 @@ func main() {
 		os.Exit(1)
 	}
 
-	// upstreamClient is the http client used to send things on to Honeycomb
-	upstreamClient := &http.Client{
-		Timeout: time.Second * 10,
-		Transport: &http.Transport{
-			Dial: (&net.Dialer{
-				Timeout: 5 * time.Second,
-			}).Dial,
-			TLSHandshakeTimeout: 6 * time.Second,
+	// upstreamTransport is the http transport used to send things on to Honeycomb
+	upstreamTransport := &http.Transport{
+		Dial: (&net.Dialer{
+			Timeout: 10 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout: 15 * time.Second,
+	}
+
+	// peerTransport is the http transport used to send things to a local peer
+	peerTransport := &http.Transport{
+		Dial: (&net.Dialer{
+			Timeout: 1 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout: 1200 * time.Millisecond,
+	}
+
+	userAgentAddition := "samproxy/" + version
+	upstreamClient, err := libhoney.NewClient(libhoney.ClientConfig{
+		Transmission: &transmission.Honeycomb{
+			MaxBatchSize:         libhoney.DefaultMaxBatchSize,
+			BatchTimeout:         libhoney.DefaultBatchTimeout,
+			MaxConcurrentBatches: libhoney.DefaultMaxConcurrentBatches,
+			PendingWorkCapacity:  libhoney.DefaultPendingWorkCapacity,
+			UserAgentAddition:    userAgentAddition,
+			Transport:            upstreamTransport,
+		},
+	})
+	if err != nil {
+		fmt.Printf("unable to initialize upstream libhoney client")
+		os.Exit(1)
+	}
+
+	peerClient, err := libhoney.NewClient(libhoney.ClientConfig{
+		Transmission: &transmission.Honeycomb{
+			MaxBatchSize:         libhoney.DefaultMaxBatchSize,
+			BatchTimeout:         libhoney.DefaultBatchTimeout,
+			MaxConcurrentBatches: libhoney.DefaultMaxConcurrentBatches,
+			PendingWorkCapacity:  libhoney.DefaultPendingWorkCapacity,
+			UserAgentAddition:    userAgentAddition,
+			Transport:            upstreamTransport,
+			// gzip compression is expensive, and peers are most likely close to each other
+			// so we can turn off gzip when forwarding to peers
+			DisableGzipCompression: true,
 		},
+	})
+	if err != nil {
+		fmt.Printf("unable to initialize upstream libhoney client")
+		os.Exit(1)
 	}
 
 	var g inject.Graph
 	err = g.Provide(
 		&inject.Object{Value: c},
 		&inject.Object{Value: lgr},
-		&inject.Object{Value: upstreamClient, Name: "upstreamClient"},
-		&inject.Object{Value: &transmit.DefaultTransmission{}},
+		&inject.Object{Value: upstreamTransport, Name: "upstreamTransport"},
+		&inject.Object{Value: peerTransport, Name: "peerTransport"},
```

Comment:
Do we still need to inject the transports if we're injecting the clients?

Reply:
Yup for now because the logger and metrics things declare their own client. Also the router declares its own HTTP client, which needs an upstream transport. This could all probably be detangled a bit more though.